### PR TITLE
Build Java without ANT

### PR DESCRIPTION
**Pull request by dkurt**
### Pull Request Readiness Checklist

Enables a path of building Java bindings without ANT

* Able to build OpenCV JAR and Docs without ANT
  ```
  --   Java:                          
  --     ant:                         NO
  --     JNI:                         /usr/lib/jvm/default-java/include /usr/lib/jvm/default-java/include/linux /usr/lib/jvm/default-java/include
  --     Java wrappers:               YES
  --     Java tests:                  NO
  ```
* Possible to build OpenCV JAR without ANT but tests still require ANT

**Merge with**: https://github.com/opencv/opencv_contrib/pull/3502

Notes:
- Use `OPENCV_JAVA_IGNORE_ANT=1` to force "Java" flow for building Java bindings
- Java tests still require Apache ANT
- JAR doesn't include `.java` source code files.

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/java/jar/CMakeLists.txt [hunk 1]
```diff
@@ -34,19 +34,41 @@ if(OPENCV_JAVADOC_LINK_URL)
...
+                    COMMAND ${CMAKE_COMMAND}
+                    -D OPENCV_JAVA_DIR="${OPENCV_JAVA_DIR}"
+                    -D OUTPUT="${OPENCV_JAVA_DIR}/java_sources"
+                    -P "${CMAKE_CURRENT_SOURCE_DIR}/list_java_sources.cmake")
+
+  get_filename_component(OPENCV_JAR_DIR ${OPENCV_JAR_FILE} DIRECTORY)
+  get_filename_component(JAR_NAME_WE ${JAR_NAME} NAME_WE)
+
+  add_jar(${the_module}_jar
+          SOURCES "@${OPENCV_JAVA_DIR}/java_sources"
```
Comment (dkurt) [inline modules/java/jar/CMakeLists.txt:65]: requires CMake 3.4, so probably need to add `cmake_minimum_required`?

Comment (opencv-alalek) [reply modules/java/jar/CMakeLists.txt:69]: OpenCV 4.x requires CMake 3.5.1 (base version for Ubuntu 16.04)

File: modules/java/CMakeLists.txt [hunk 2]
```diff
@@ -3,7 +3,9 @@ if(OPENCV_INITIAL_PASS)
   add_subdirectory(generator)
 endif()
 
-if(APPLE_FRAMEWORK OR WINRT OR NOT PYTHON_DEFAULT_AVAILABLE OR NOT (ANT_EXECUTABLE OR ANDROID_PROJECTS_BUILD_TYPE STREQUAL "GRADLE")
+if(APPLE_FRAMEWORK OR WINR
```
Comment (asmorkalov) [inline modules/java/CMakeLists.txt:6]: Typo in WINRT.

**Review by opencv-alalek — APPROVED** [review]
- [x] Validated with "Java" JAR + "ANT" Java tests

**Review by asmorkalov — APPROVED** [review]

**Pull request merged**